core/network/besu/contracts/interop/manageAssetAny.sol

```diff
@@ -57,18 +58,65 @@ contract AssetExchangeContract is ERC1155Holder {
         bytes32 indexed lockContractId
     );
 
-    // The sender locks a  asset with a hash lock and an expiry time
+    function bytesToAddress(bytes memory bys)
+        public
+        pure
+        returns (address addr)
+    {
+        assembly {
+            addr := mload(add(bys, 20))
+        }
+    }
+
+    function bytesTobytes32(bytes memory bys)
+        public
+        pure
+        returns (bytes32 res)
+    {
+        assembly {
+            res := mload(add(bys, 32))
+        }
+    }
+
+    function stringToUint(string memory s) internal returns (uint256 result) {
+        bytes memory b = bytes(s);
+        uint256 i;
+        result = 0;
+        for (i = 0; i < b.length; i++) {
+            uint256 c = uint256(uint8(b[i]));
+            if (c >= 48 && c <= 57) {
+                result = result * 10 + (c - 48);
+            }
+        }
+    }
+
     function lockAsset(
-        address receiver,
+        bytes memory rawParams,
         address assetContract,
         uint256 amount,
-        bytes32 hashLock,
-        uint256 expirationTime,
-        uint256 tokenId,
+        bytes memory lockinfo,
         bytes memory data
     ) external returns (bytes32 lockContractId) {
+        AssetExchangeAgreement memory params;
+        (, , params) = AssetExchangeAgreementCodec.decode(
+            0,
+            rawParams,
+            uint64(rawParams.length)
+        );
+        AssetLockHTLC memory assetInfo;
+        (, , assetInfo) = AssetLockHTLCCodec.decode(
+            0,
+            lockinfo,
+            uint64(lockinfo.length)
+        );
+        HashMechanism hashMechanism = assetInfo.hashMechanism;
+        bytes32 hashLock = bytesTobytes32(assetInfo.hashBase64);
+        if (hashMechanism == HashMechanism.SHA256) {
+            hashLock = sha256(abi.encodePacked(hashLock));
```

Comment:
I don't get what are you trying to do here, `hashLock` itself is the hash, why are you computing hash of the hash?